### PR TITLE
HPCC-13895 Fix problems with embedded datasets in nested rows

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday gavin.halliday@lexisnexis.com

### CODE REVIEW

**Comment by ghalliday** [comment]
@jamienoss please review.

File: ecl/hqlcpp/hqliproj.ipp [hunk 1]
```diff
@@ -158,6 +158,7 @@ public:
 protected:
     bool contains(IHqlExpression & field) const;
     bool contains(IAtom * name) const; // debugging only
+    IHqlExpression * findByName(IAtom * name) const; // debugging only
```
Comment (jamienoss) [inline ecl/hqlcpp/hqliproj.ipp]: 'debugging only'? I looked through a level or two up but couldn't find anything that looked like conditional code to suggest that this was called in a debug context only. I guess this refers to something else?

Comment (ghalliday) [inline ecl/hqlcpp/hqliproj.ipp]: Yes the comment shouldn't be there.  I will remove.

**Comment by jamienoss** [comment]
@ghalliday finished review with only one trivial comment.

**Comment by ghalliday** [comment]
@richardkchapman updated to remove the comment

**Pull request merged**